src/databricks/labs/ucx/recon/base.py

```diff
@@ -99,7 +106,12 @@ def compare_schema(self, source: TableIdentifier, target: TableIdentifier) -> Sc
 
 class DataComparator(ABC):
     @abstractmethod
-    def compare_data(self, source: TableIdentifier, target: TableIdentifier) -> DataComparisonResult:
+    def compare_data(
+        self,
+        source: TableIdentifier,
+        target: TableIdentifier,
+        row_comparison: bool,
```

Comment:
```suggestion
        compare_rows: bool,
```
might express the intent better, right?..